packages/suite/src/components/suite/DeviceDisplay/parseTextToPagesAndLines.ts

```diff
@@ -0,0 +1,84 @@
+import { DeviceModelInternal } from '@trezor/connect';
+import {
+    MAX_ROWS_PER_PAGE,
+    MAX_CHARACTERS_ON_ROW,
+    CHARACTER_OFFSET_FOR_CONTINUES_ARROW,
+    CHARACTER_OFFSET_FOR_NEXT_ARROW,
+} from '../../../constants/suite/device';
+
+export type ParseTextToLinesParams = {
+    device: DeviceModelInternal;
+    text: string;
+};
+
+export type ResultRow = { text: string };
+export type ResultPage = { rows: ResultRow[] };
+
+export type ParseTextToLinesResult = {
+    pages: ResultPage[];
+    hasNextIcon: boolean;
+};
+
+export const parseTextToPagesAndLines = ({
+    device,
+    text,
+}: ParseTextToLinesParams): ParseTextToLinesResult => {
+    const rowsPerPage = MAX_ROWS_PER_PAGE[device];
+    const charsPerRow = MAX_CHARACTERS_ON_ROW[device];
+    const offsetForContinuesArrows = CHARACTER_OFFSET_FOR_CONTINUES_ARROW[device];
+    const offsetForNextArrows = CHARACTER_OFFSET_FOR_NEXT_ARROW[device];
+
+    const charsOnLastPage = rowsPerPage * charsPerRow - offsetForNextArrows;
+
+    let pageIndex = 0;
+    let remaining = text;
+    let pages: ResultPage[] = [];
+
+    while (true) {
+        const isFirstPage = pageIndex === 0;
+        const isLastPage = remaining.length <= charsOnLastPage;
+
+        let rowIndex = 0;
+        const rows: ResultRow[] = [];
+
+        while (rowIndex < rowsPerPage) {
+            const isFirstRow = rowIndex === 0;
+            const isLastRow = rowIndex === rowsPerPage - 1;
+
+            const isShorterFirstRow = isFirstRow && !isFirstPage;
+            const isShorterLastRow = isLastRow && !isLastPage;
+
+            const charsShorter =
+                // eslint-disable-next-line no-nested-ternary
```

Comment:
We shouldn't violate our own rules :)

Reply:
Are you sure you want this? https://github.com/trezor/trezor-suite/pull/11251/commits/44b60f9d56c668841496506df35d13d3b0345447#diff-241be0b07cc1eadc4831a5a3e7b4296b2c552ba6975c34c3e9ded93a8b08df22R22

Do you have better alternative?

Reply:
Or maybe this? 
![image](https://github.com/trezor/trezor-suite/assets/152899911/d4b1e996-639d-4b0a-83f4-d01b5f6e976c)


Still IMO the nested ternar is nicier.

Reply:
Not saying the original solution was too bad. We can discuss the rule on code improvements.

Reply:
OK I will revert to nested ternar ok?

Reply:
No no I mean let's keep this unless we decide to abandon the rule.

Reply:
So you are ok with more complicated, less readable solution rather then ignoring the rule in this case?

Reply:
Yes, I don't like disabling eslint inline because it makes it feel meaningless. If we find some rule inadequate, we should change the rule first. Although this could be a topic for code improvements discussion by itself.

Reply:
ok, I am more on the side of braking rules that don't serve well, but ok